doc/contributing.rst

```diff
@@ -122,17 +128,58 @@ want to clone your fork to your machine::
 This creates the directory `xarray` and connects your repository to
 the upstream (main project) *xarray* repository.
 
+Creating a branch
```

Comment:
In a later PR, IMO we should link out to the Github docs more for the  general stuff. They put a bunch of effort in to making it easy. E.g. https://docs.github.com/en/get-started/quickstart/github-flow